### PR TITLE
Admin page to see team lunch orders

### PR DESCRIPTION
**Pull request by morganbr**
Shows a summary report and individual team lunch orders. Includes defaulting for lunches that were not chosen

### CODE REVIEW

**Comment by tabascq** [comment]
Can you explain more precisely what this is displaying? I can't follow the query logic through all the grouping etc. Example questions:

- Does it show orders per team? Does it show how many of those orders are default?
- Does it show cumulative orders for the event? Does it show how many of those orders are default?
- Does it keep track of teams that have ordered more pizzas than they are permitted, due to team member cancellations? (Is this even possible?)

Some sample output would help.

File: ServerCore/Pages/Swag/TeamReport.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        public List<SwagView> SwagViews { get; set; }
+        public List<SummaryItem> Summary { get; set; }
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            // Fetch lunches the team chose
+            SwagViews = await (from teamLunch in _context.TeamLunch
+                         join team in _context.Teams on teamLunch.TeamId equals team.ID
```
Comment (digit01Wave) [inline ServerCore/Pages/Swag/TeamReport.cshtml.cs:51]: question: Does this need a join if Team is already a virtual navigation property of TeamLunch?

File: ServerCore/Pages/Swag/TeamReport.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            // Find out how many lunches weren't ordered and should be default:
+            // 1. Find out how many got ordered
+            var teamLunchCounts = from team in SwagViews
+                             group team by team.TeamId into teamGroup
+                             select new { TeamId = teamGroup.Key, Count = teamGroup.Count() };
+
+            // 2. Find out how many lunches the team is allowed to order
+            var teamEligibleMembers = await (from team in _context.Teams
+                                      join member in _context.TeamMembers on team equals member.Team
+                                      join player in _context.PlayerInEvent on member.Member equals player.Player
```
Comment (digit01Wave) [inline ServerCore/Pages/Swag/TeamReport.cshtml.cs:71]: [question]: Does this also need a join if Player is a navigation property of TeamMembers?

**Comment by morganbr** [comment]
Here's a sample screenshot where a team has chosen one pizza and has failed to choose another and is getting the default of cheese
image

**Comment by morganbr** [comment]
>     * Does it show orders per team? Does it show how many of those orders are default?
It shows totals and each team's orders

>     * Does it show cumulative orders for the event? Does it show how many of those orders are default?
Yes and yes

>     * Does it keep track of teams that have ordered more pizzas than they are permitted, due to team member cancellations? (Is this even possible?)
It could show that, but doesn't currently. Do we need it?

> Some sample output would help.
See above

File: ServerCore/Pages/Swag/TeamReport.cshtml.cs (hunk 1 above)
Comment (morganbr) [reply ServerCore/Pages/Swag/TeamReport.cshtml.cs:51]: Using the navigation property also does a join, this is just more explicit

File: ServerCore/Pages/Swag/TeamReport.cshtml.cs (hunk 2 above)
Comment (morganbr) [reply ServerCore/Pages/Swag/TeamReport.cshtml.cs:71]: Similarly, these navigation properties do the same kind of join

**Review by asyasky — APPROVED** [review]

**Pull request merged**